src/index.js

```diff
@@ -249,145 +243,151 @@ class TerserPlugin {
         []
       );
       const files = [].concat(additionalChunkAssets).concat(chunksFiles);
-      const tasks = [];
 
-      files.forEach((file) => {
-        if (!matchObject(file)) {
-          return;
-        }
+      function* tasks() {
```

Comment:
Move this to class method, for better readable

Reply:
Good idea! I moved this to a separate method as part of a separate commit fa36e0f